IPs/RSKIP144.md

```diff
@@ -19,47 +19,36 @@ RSK processes transactions from blocks one by one, in the specified order. This
+A new field ``partitionEnds`` is added to the block header. This field consists of an array of shorts that indicates at which position in the transaction list each partition ends. For example, in a block with 10 transactions, ``partitionEnds = [3, 6]`` indicates that the first partition contains transactions 0, 1 and 2; the second partition contains transactions 3, 4 and 5; and the last partition contains transactions 6 to 9. Values in ``partitionEnds`` must be greater than 0 and in ascending order. An empty ``partitionEnds`` indicates that all transactions go in the sequential partition, and the REMASC transaction is always the last transaction of the sequential partition. The maximum number of parallel partitions that the miner can specify is equal to the minimum number of cores required to run the RSK node.
 
-Always at the end of processing a transaction, the new reads/writes are copied to the global write/reads maps for efficiency.  This procedure repeats until there is no more gas left to consume or there are no more valid transactions to add. 
+During execution, when a storage key is read, it is marked with the thread index in a per-partition readMap. When a key is written, the key is marked in a per-partition writeMap map. When all transactions of a certain partition have been processed, the writeMap is scanned. When all parallel partitions have finished processing, the readMaps and writeMaps are "merged". This requires efficient maps that enable traversing the keys in ascending lexicographic order. If a key belongs to a writeMap and a readMap of another partition, then the block is considered invalid. If a key belongs to a writeMap and a writeMap of another partition, then the block is also considered invalid. Recursive deletes must be correctly and efficiently handled. Miners are incentivized to produce a valid and efficient partition because by doing so their blocks spread faster over the network.
 
-Special treatment may be needed for recursive deletes, as used in suicides. 
+When a miner executes transactions to create an (unsolved) block, the miner must execute the transactions serially and decide which partition the transaction belongs to. The block gas limit is replaced by a per-partition gas limit. We say that a transaction is *connected* to another transaction if:
```

Comment:
How the miner creates the partitions is not strictly a part of the consensus specification. It's part of the implementation. Therefore I suggest that all the description on how the miner creates the block is moved to another section: "Suggested Miner Implementation"

Reply:
ok!